Selectors/src/main/java/io/doist/recyclerviewext/choice_modes/Selector.java

```diff
@@ -50,16 +50,44 @@ public void toggleSelected(long id) {
 
     public abstract void clearSelected();
 
+    /**
+     * @deprecated use {@link #addOnSelectionChangedListener(OnSelectionChangedListener)}
+     * to add a listener or {@link #removeAllOnSelectionChangedListeners()} to remove all listeners.
+     */
+    @Deprecated
     public void setOnSelectionChangedListener(@Nullable OnSelectionChangedListener observer) {
-        mObserver = observer;
+        removeAllOnSelectionChangedListeners();
+        if (observer != null) {
+            addOnSelectionChangedListener(observer);
+        }
+    }
+
+    public void addOnSelectionChangedListener(@NonNull OnSelectionChangedListener observer) {
+        mObservers.add(observer);
+    }
+
+    public void removeOnSelectionChangedListener(@NonNull OnSelectionChangedListener observer) {
+        mObservers.remove(observer);
+    }
+
+    public void removeAllOnSelectionChangedListeners() {
+        mObservers.clear();
+    }
+
+    protected void onSelectionChanged(long[] selectedIds, long[] previousSelectedIds) {
+        for (int i = 0; i < mObservers.size(); ++i) {
```

Comment:
Just curious why you used `++i` instead of `i++`. 😅

Reply:
There's no really a strong reason for that 😄